src/sdk/pynni/nni/analysis_utils/sensitivity/torch/sensitivity_analysis.py

```diff
@@ -0,0 +1,257 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import os
+import copy
+import csv
+import logging
+from collections import OrderedDict
+import matplotlib
+import matplotlib.pyplot as plt
+from matplotlib.lines import Line2D
+
+import numpy as np
+import torch.nn as nn
+
+from nni.compression.torch import LevelPruner
+from nni.compression.torch import L1FilterPruner
+from nni.compression.torch import L2FilterPruner
+
+# use Agg backend
+matplotlib.use('Agg')
+SUPPORTED_OP_NAME = ['Conv2d', 'Conv1d']
+SUPPORTED_OP_TYPE = [getattr(nn, name) for name in SUPPORTED_OP_NAME]
+
+logger = logging.getLogger('Sensitivity_Analysis')
+logger.setLevel(logging.INFO)
+
+
+class SensitivityAnalysis:
+    def __init__(self, model, val_func, sparsities=None, prune_type='l1', early_stop=1.0):
+        """
+        Perform sensitivity analysis for this model.
+        Parameters
+        ----------
+            model:
```

Comment:
the docstring format of the input arguments is:
arg : type
you missed the type field.